### ISSUE TITLE
Improve documentation about JDK > 11

### ISSUE THREAD

**Issue by edmondop** [unitycatalog/unitycatalog#4]
At the moment, building the project with JDK 21 is not supported since SBT 1.6 support up to Java 17, see

https://docs.scala-lang.org/overviews/jdk-compatibility/overview.html

Trying to build on SDK 21 results in an error

```
copying runtime jar...
[info] welcome to sbt 1.6.1 (Amazon.com Inc. Java 21.0.2)
error:
  bad constant pool index: 0 at pos: 48461
     while compiling: <no file>
        during phase: globalPhase=<no phase>, enteringPhase=<some phase>
     library version: version 2.12.15
    compiler version: version 2.12.15
  reconstructed args: -classpath /Users/edmondop/.sbt/boot/scala-2.12.15/lib/scala-library.jar -Yrangepos

  last tree to typer: EmptyTree
       tree position: <unknown>
            tree tpe: <notype>
              symbol: null
           call site: <none> in <none>

== Source file context for tree position ==

error:
  bad constant pool index: 0 at pos: 48461
```

Although this is a SBT problem, it's probably worth documenting it

**Comment by tdas** [comment]
The sbt just got upgraded to 1.10. does that work for you?

**Comment by edmondop** [comment]
> The sbt just got upgraded to 1.10. does that work for you?

Thanks @tdas @rxin has merged #11 . I think just improving the docs is fine, in the updated README there is a pointer between java versions and SBT versions compatibility

**Comment by edmondop** [comment]
Fixed via #11
